### DISCUSSION TITLE
UI simulator run failure

### DISCUSSION THREAD

**Discussion by baymac** [KeystoneHQ/keystone3-firmware#1373] [Q&A] [answered]
I am using Mac M1 Pro, Mac OS 14.1, followed the docs but facing an error. Not very familiar with rust, can I please get some help to understand more about the error described below? 

`python3 build.py -o simulator`, throws this error:

```
ERROR: Couldn't execute `cargo metadata` with manifest "/Users/parichay/minswap/keystone3-firmware/rust/rust_c/Cargo.toml": Metadata(Output { status: ExitStatus(unix_wait_status(25856)), stdout: "", stderr: "warning: patch for `bitcoin` uses the features mechanism. default-features and features will not take effect because the patch dependency does not support this mechanism

error: no matching package named `cardano-serialization-lib` found

location searched: https://git@github.com/KeystoneHQ/cardano-serialization-lib.git?tag=keystone-0.1.3#370350ec

required by package `app_cardano v0.1.0 (/Users/parichay/minswap/keystone3-firmware/rust/apps/cardano)`

    ... which satisfies path dependency `app_cardano` (locked to 0.1.0) of package `common_rust_c v0.1.0 (/Users/parichay/minswap/keystone3-firmware/rust/rust_c/src/common)`\n" })

ERROR: Couldn't generate bindings for /Users/parichay/minswap/keystone3-firmware/rust/rust_c.

make[2]: *** [rust/rust_c/CMakeFiles/rust_c] Error 1

make[1]: *** [rust/rust_c/CMakeFiles/rust_c.dir/all] Error 2

make: *** [all] Error 2

```

Same error on branches `master` and `release/v1.7.0`.

**Comment by soralit** [comment]
This might be related to your network status, because this dependency is fetched from github.

**Comment by baymac** [comment]
I am able to install other packages, ping github.com across multiple networks. I can try it on a different machine later to verify. Btw I am wondering if I can side load firmware into the hardware directly? It would be easier to work with.

**Comment by soralit** [comment]
If you mean building firmware by yourself and load it into your device, I'm sadly to say no. The released firmwares are signed by us and verified by the hardware. 
And the built simulator could only run on your PC.

**Comment by baymac** [comment]
I tried resolving the previous error, and that issue is now fixed. However, I’m facing new errors that I can't make sense of. I've shared the logs at [link](https://gist.github.com/baymac/cf809b3892e3364a9a24759a58358b1f) for reference. If you have a solution please let me know.

We might as well move this issue to discussions tab.

**Comment by soralit** [comment]
seems you didn't get the submodule, please follow the command in this section: https://github.com/KeystoneHQ/keystone3-firmware?tab=readme-ov-file#macos
`git -c submodule.keystone3-firmware-release.update=none submodule update --init --recursive`

**Reply by baymac** [reply]
Its fixed wow! I can compile the binary but get this error when I run it:

```
startfile not exist: ./ui_simulator/assets/device_setting.json
lv_fs_open failed C:/assets/device_setting.json ret = 12 line = 100
assert,file=/Users/parichay/minswap/keystone3-firmware/keystone3-firmware/src/device_settings.c
```

**Reply by soralit** [reply]
do you run on a Windows device?

**Reply by baymac** [reply]
I am running on MacOS 14.1

**Reply by baymac** [reply]
@soralit Could you help me with running the firmware? There is no other way we could integrate with Keystone. I have tried following the docs but it doesn't work on my machine. Is it possible for you to dockerize the simulator or test it on a server instance for keystone.

**Reply by baymac** [reply, marked as answer]
I was able to make it work. Uninstall rust and all libs and restarted steps again - build succeeded.

To avoid `startfile not exist: ./ui_simulator/assets/device_setting.json` error, instead of opening the bin file I ran `build/simulator` command on terminal.
